vme/src/descriptor_data.cpp

```diff
@@ -300,7 +300,7 @@ const char *descriptor_data::getLastCommand() const
 
 void descriptor_data::setLastCommand(const char *value)
 {
-    strncpy(last_cmd, value, sizeof(last_cmd));
+    strncpy(last_cmd, value, sizeof(last_cmd) - 1);
```

Comment:
Warning here that strncpy will not null terminate if size matches - so do 1 less

Reply:
Great catch. But don't we ahve to set 

last_cmd[sizeof(last_cmd)-1] = 0;

Otherwise, if value is longer than last_cmd, you're not going to be guaranteed that last_cmd[29] is 0. Unless ofc we know for sure that last_cmd[29] was set to 0 somewhere else. Just seems like a nice extra precaution. strncpy is tricky.

Reply:
Yeah you are right, I'll add those to the strncpy's